staff/daniel-toledo/e-commerce-protos/scripts/cart.test.js

```diff
@@ -0,0 +1,133 @@
+suite('Cart');
+
+test('creats a Cart and add products', function(){
+    var cart = new Cart;
+
+    cart.add('mobile');
+    cart.add('mobile2');
+    cart.add('laptop');
+    cart.add('desktop');
+
+
+    assert(cart.cartList.toString()===['mobile', 'mobile2','laptop','desktop'].toString(), 'Unexpected value')
+
+});
+
+test('total price', function(){
+    var cart = new Cart;
+
+    var socks = new Socks('Calvin Klein', 42, 'black', 9.99);
+    var tanga = new Tanga('Wicked Weasel', 32, 'red', 29.95);
+    var bra = new Bra('Calvin Klein', 'push-up', 80, 'cream', 39);
+
+    cart.add(socks);
+    cart.add(tanga);
+    cart.add(bra);
+
+    res=cart.totalPrice()
+
+    expected=78.94
+
+    assert(res===expected, 'Unexpected value')
+
+});
+
+test('Number of items', function(){
+    var cart = new Cart;
+
+    cart.add('mobile');
+    cart.add('mobile2');
+    cart.add('laptop');
+    cart.add('desktop');
+
+    res=cart.numberOfItems()
+
+
+    assert(res===4, 'Unexpected value')
+
+});
+
+test('more expensive', function(){
+    var cart = new Cart;
+
+    var socks = new Socks('Calvin Klein', 42, 'black', 9.99);
+    var tanga = new Tanga('Wicked Weasel', 32, 'red', 29.95);
+    var bra = new Bra('Calvin Klein', 'push-up', 80, 'cream', 39);
+
+    cart.add(socks);
+    cart.add(tanga);
+    cart.add(bra);
+
+    res=cart.mostExpensive().price
+
+    expected=39
+
+    assert(res===expected, 'Unexpected value')
+
+});
+
+test('cheapest', function(){
+    var cart = new Cart;
+
+    var socks = new Socks('Calvin Klein', 42, 'black', 9.99);
+    var tanga = new Tanga('Wicked Weasel', 32, 'red', 29.95);
+    var bra = new Bra('Calvin Klein', 'push-up', 80, 'cream', 39);
+
+    cart.add(socks);
+    cart.add(tanga);
+    cart.add(bra);
+
+    res=cart.cheapest().price
+
+    expected=9.99
+
+    assert(res===expected, 'Unexpected value')
+
+});
+
+test('product by price range', function(){
+    var cart = new Cart;
+
+    var socks = new Socks('Calvin Klein', 42, 'black', 9.99);
+    var tanga = new Tanga('Wicked Weasel', 32, 'red', 29.95);
+    var bra = new Bra('Calvin Klein', 'push-up', 80, 'cream', 39);
+
+    cart.add(socks);
+    cart.add(tanga);
+    cart.add(bra);
+
+    res=cart.productsByPriceRange(9,30)
+
+    expected=[{size: 42, color: "black", brand: "Calvin Klein", price: 9.99},
+              {size: 32, color: "red", brand: "Wicked Weasel", price: 29.95}]
+
```

Comment:
It's the same, and easier to read if you write: 
```javascript
expected=[Socks, Tanga]
```

Reply:
Uauu really? Nice!!
 I didn't know it :grin:

Reply:
Ahhh but I am thinking... if you have two Tangas, with different properties, it would not find the differences. it would consider that both are Tangas

Reply:
You should check what effect have the toString() on res and on expected. And also, the expected array should be filled of instances of products, not simple arrays!

Reply:
Done!!
`expect (res instanceof Bra, 'res should have right instanceof')`
and more.... I did it also in other cases that I had the same problem